### DISCUSSION TITLE
How to add command line argument for particular tests subdirectory?

### DISCUSSION THREAD

**Discussion by ghost** [pytest-dev/pytest#11144] [General]
Hi,

I have a subset of tests where I want to apply a command line argument specific to those tests.  The argument defines a file that should be loaded by a fixture.

I've been trying to use `addinivalue_line` from within `pytest_configure` but this doesn't seem to work.  How is this supposed to be used?

As a simple example of what I'm trying to do here is an example `conftest.py` at the top level of my tests directory:

```
def pytest_addoption(parser):
    parser.addoption(
        "--ping",
        action="store_true",
        help="ping pong"
    )
    parser.addoption(
        "--pong",
        help="store a word"
    )

def pytest_configure(config):
    if config.getoption("ping"):
        config.addinivalue_line("addopts", "--pong=ball")
```

and a matching test:

```
def test_this(request):
    log.info(request.config.inicfg)
    log.info(f"pong={request.config.getoption('pong')}")
```

I'm expecting that if I add the `--ping` CLI arg that I then get `pong` set.

**Comment by RonnyPfannschmidt** [comment]
pytest_configure happens way after addopts is considered

there is no builtin way to do what you outline

please outline the intended use-case more comprehensible as the minified example is not enough to make a good suggestion

**Comment by ghost** [comment]
Ah okay, I see.

I have a fixture that uses a YAML configuration file for a particular feature and this is passed on the command line.  So my tests are organised in a hierarchy where only one directory needs to have this command line argument passed and I was trying to dynamically add this to the ini file using `conftest.py`.

What's an alternate way of specifying command line args dynamically per directory?

**Reply by RonnyPfannschmidt** [reply]
There isn't a way to do that's,cli args are done before

**Reply by ghost** [reply]
I've just done it using `pytest_cmdline_preparse` but I get a warning that this function is deprecated.  Has it been replaced with something else?  The documentation here also provides a way to add command line args:  https://docs.pytest.org/en/latest/example/simple.html#dynamically-adding-command-line-options
